app/controllers/application_controller.rb

```diff
@@ -26,4 +29,26 @@ def set_i18n_locale_from_params
         end
       end
     end
+
+    def set_counter
+      if(session[:user_id])
+        user_counter = Counter.find_by(user_id:session[:user_id])
+        user_counter.count += 1
+        @counter = user_counter.count
+        user_counter.save
+      end
+    end
+    
+    def check_last_activity
+      if(session[:user_id])
+        user = User.find(session[:user_id])
```

Comment:
User.find(session[:user_id]) is being done again and again, so you can save this in @current_user in authorize method itself and use @current_user wherever you need to get the logged in user